clang/test/CodeGenSYCL/esimd_metadata2.cpp

```diff
@@ -0,0 +1,42 @@
+// RUN: %clang_cc1 -disable-llvm-passes -triple spir64-unknown-unknown-sycldevice -fsycl -fsycl-is-device -fsycl-explicit-simd -S -emit-llvm %s -o - | FileCheck %s --check-prefixes CHECK,CHECK-ESIMD
+
+// In ESIMD mode:
+// 1. Attribute !intel_reqd_sub_group_size !1 is added
+//    for kernels with !sycl_explicit_simd
+// 2. Attribute !sycl_explicit_simd is propagated to all the
+//    callees of ESIMD kernel.
+
+__attribute__((sycl_device)) void shared_func_decl();
+__attribute__((sycl_device)) void shared_func() { shared_func_decl(); }
+
+__attribute__((sycl_device)) __attribute__((sycl_explicit_simd)) void esimd_func() { shared_func(); }
+
+// CHECK-NOT: !sycl_explicit_simd
+// CHECK-NOT: !intel_reqd_sub_group_size
```

Comment:
What are the CHECK-NOTs for?

Reply:
They check that there attributes are propagated only to a limited set of functions, i.e. no other function is accidentally marked.

Reply:
There is one more function `ESIMDFunctor::operator()`. I added it and removed `CHECK-NOT`s. For now we do not support mixing SYCL and ESIMD kernels, so I decided not to add additional functions to the test, since it will only confuse developers.